chalicelib/app_utils.py

```diff
@@ -73,14 +75,15 @@ def init_environments(env='all'):
     return environments
 
 
-def init_connection(environ):
+def init_connection(environ, environments=None):
     """
     Initialize the fourfront/s3 connection using the FSConnection object
     and the given environment.
     Returns an FSConnection object or raises an error.
     """
     error_res = {}
-    environments = init_environments()
+    if environments is None:
```

Comment:
Good catch that this is not ideal. However, I don't think this is the way to fix it. What about:
```
def init_connection(environ):
    environments = init_environments(environ)   # will return empty dict if not valid
   ...
```
The rest of the code would be mostly the same, only thing is the error message would have to change because you're not guaranteed to have all environments available. To keep this functionality, you could do:
```
if environ not in environments:
    environments = init_environments('all')  # same as init_environments()
    # continue with error message
    ...
```

Reply:
In the second code block, why call `init_environments` at all there if you're just going to raise an exception immediately after? Should it check again if `all` populated `environ`?